cmd/limactl/start.go

```diff
@@ -529,3 +529,55 @@ func startBashComplete(cmd *cobra.Command, _ []string, _ string) ([]string, cobr
 	compTmpl, _ := bashCompleteTemplateNames(cmd)
 	return append(compInst, compTmpl...), cobra.ShellCompDirectiveDefault
 }
+
+// startInteractive starts the instance after a confirmation prompt.
+//
+// If newInst is set to true, this function creates the instance too.
+//
+// If mandatory is set to true, and the answer to the confirmation prompt is No,
+// the function returns an error that contains an instruction to start the instance manually.
+func startInteractive(cmd *cobra.Command, instName string, newInst, mandatory bool) (bool, error) {
+	flags := cmd.Flags()
+	tty, err := flags.GetBool("tty")
+	if err != nil {
+		return false, err
+	}
+	errS := fmt.Sprintf("instance %q is stopped, run `limactl start %s` to start the instance", instName, instName)
+	if newInst {
+		errS = fmt.Sprintf("instance %q does not exist, run `limactl create %s` to create a new instance", instName, instName)
+	}
+	if !tty {
+		if mandatory {
+			return false, errors.New(errS)
+		}
+		return false, nil
+	}
+	logrus.Error(errS)
+
+	message := fmt.Sprintf("Do you want to start the instance %q now?", instName)
+	if newInst {
+		message = fmt.Sprintf("Do you want to create and start the instance %q using the default template now?", instName)
+	}
+	ans, err := uiutil.Confirm(message, true)
+	if err != nil {
+		return false, err
+	}
+	if !ans {
+		if mandatory {
+			return ans, errors.New(errS)
+		}
+		return ans, nil
+	}
+	if newInst {
+		rootCmd := cmd.Root()
+		// The create command shows the template chooser UI, etc.
```

Comment:
I would have expected that the template would default to the instance name, if a template existed with that name. Otherwise it should show an error and only then default to `default`.

```console
$ l shell alpine
ERRO[0000] instance "alpine" does not exist, run `limactl create alpine` to create a new instance
? Do you want to create and start the instance "alpine" using the default template now?
```

This should say "using the alpine template".

Reply:
I agree that the `mandatory` mechanism is too complex/confusing. It also results in the same error being displayed twice, first as `ERRO` and then as `FATA`:

```console
$ l shell alpine
ERRO[0000] instance "alpine" does not exist, run `limactl create alpine` to create a new instance
? Do you want to create and start the instance "alpine" using the default template now? No
FATA[0002] instance "alpine" does not exist, run `limactl create alpine` to create a new instance
```

I think the `mandatory` argument can just be removed, and the function always just returns `nil` when the user elected not to create/start the instance.

The caller is already calling `store.Inspect()` afterwards, and can then simply exit if the instance isn't in running state.